staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -299,6 +300,22 @@ func (s *store) GuaranteedUpdate(
 
 		ret, ttl, err := s.updateState(origState, tryUpdate)
 		if err != nil {
+			// It's possible we were working with stale data
+			if mustCheckData && apierrors.IsConflict(err) {
```

Comment:
Lets make this logic a function since its the same as above https://github.com/kubernetes/kubernetes/pull/43152/files#diff-0f11327dd3daafdfe8e326e895616f66R284

Reply:
I'm not sure that's entirely necessary given that it's only used in 2 places. If it's 3+, definitely... but if you want it, I can do it. Have a suggestion for a name?

Reply:
It's going to end up having to be

```
func (s *store) yourNameHere(ctx context.Contect, key string, v reflect.Value, ignoreNotFound bool) (*objState, error)
```

Not convinced we need it.

Reply:
Ha ok I do see it used in 3 places